src/main/java/dev/dbos/transact/database/SystemDatabase.java

```diff
@@ -422,8 +426,20 @@ private <T> T handleExistingResult(StepResult result, String functionName) {
         }
     }
 
-    public static HikariDataSource createDataSource(DBOSConfig config, String dbName) {
+    public static HikariDataSource createDataSource(String url, String user, String password, int poolSize, int timeout) {
```

Comment:
Why specify user and password separately from url?

Reply:
that's how JDBC works. username and password are not passed in their normal locations in the URL. You can pass them in the URL but as [query params](https://jdbc.postgresql.org/documentation/use/#connection-parameters).

We may want to add a compat layer so that you specify PG URLs as we do in DBOS Cloud and then transform them into the equivalent JDBC URL, but I think that should be a separate PR